### PR TITLE
Add an initial rbs template to `bundle gem` skeleton

### PR DESCRIPTION
**Pull request by hsbt**
## What was the end-user or developer problem that led to this PR?

[rbs](https://github.com/ruby/rbs) is `RBS is a language to describe the structure of Ruby programs`. It's good to add it boilerplate of `bundle gem` command.

## What is your fix for the problem, implemented in this PR?

Added the initial rbs file for `newgem`.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [x] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [ ] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

File: bundler/lib/bundler/templates/newgem/sig/newgem.rbs.tt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<%- config[:constant_array].each_with_index do |c, i| -%>
+<%= "  " * i %>module <%= c %>
+<%- end -%>
+<%= "  " * config[:constant_array].size %>VERSION: String
+<%= "  " * config[:constant_array].size %># See the writing guide ob rbs: https://github.com/ruby/rbs#guides
```
Comment (pocke) [inline bundler/lib/bundler/templates/newgem/sig/newgem.rbs.tt:5]:
```suggestion
<%= "  " * config[:constant_array].size %># See the writing guide of rbs: https://github.com/ruby/rbs#guides
```

**Review by deivid-rodriguez — COMMENTED** [review]
Show we make this opt-out, through `--no-rbs` or something?

**Comment by hsbt** [comment]
I think we don't need to make optional because Matz introduce Type is the main feature of Ruby 3 or later. 

@pocke @soutaro @mame Do you have any opinion?

**Comment by hsbt** [comment]
I will ask this to @matz at the next Ruby office hour.

**Comment by matz** [comment]
I agree with the change. If you want to opt-out `rbs`, just remove `newgem.rbs` manually.

**Comment by deivid-rodriguez** [comment]
Alright, let's do this then!

**Review by simi — APPROVED** [review]

**Comment by deivid-rodriguez** [comment]
I rebased it just in case and set if for automerge after CI.

**Comment by hsbt** [comment]
@matz @deivid-rodriguez Thanks. I will confirm this with ruby master tomorrow.

**Pull request merged**